### PR TITLE
Code of conduct

### PR DESCRIPTION
**Pull request by profxj**
As titled.

### CODE REVIEW

File: doc/codeconduct.rst [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+This code of conduct applies to all community situations online and offline, including
+mailing lists, forums, social media, conferences, meetings, associated social events,
+and one-to-one interactions.
+
+Any related activity or project organized by members of the PypeIt
+community, including affiliated packages, are welcome to have their own codes
+of conduct, but agree to also abide by the present code of conduct.
+
+Parts of this code of conduct have been adapted from
```
Comment (astronomeara) [inline doc/codeconduct.rst]: need to cite where this is.

File: doc/codeconduct.rst [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+-    We pledge to welcome questions and answer them respectfully, paying particular attention to those new to the community. We pledge to provide respectful criticisms and feedback in forums, especially in discussion threads resulting from code contributions.
+-    We pledge to be conscientious of the perceptions of the wider community and to respond to criticism respectfully. We will strive to model behaviors that encourage productive debate and disagreement, both within our community and where we are criticized. We will treat those outside our community with the same respect as people within our community.
+-    We pledge to help the entire community follow the code of conduct, and to
+     not remain silent when we see violations of the code of conduct.
+     We will take action when members of our community violate this code
+     such as contacting confidential@pypeit.ucolick.org (all emails sent to this address
+     will be treated with the strictest confidence)
+     or talking privately with the person.
+
+This code of conduct applies to all community situations online and offline, including
```
Comment (astronomeara) [inline doc/codeconduct.rst]: community -> PypeIt community?

File: doc/codeconduct.rst (hunk 1 above)
Comment (profxj) [reply doc/codeconduct.rst]: which "this"?

File: doc/codeconduct.rst (hunk 2 above)
Comment (profxj) [reply doc/codeconduct.rst]: edited

File: doc/codeconduct.rst (hunk 1 above)
Comment (astronomeara) [reply doc/codeconduct.rst]: sorry, the PSF code of conduct.  I highlighted the wrong line.

Comment (profxj) [reply doc/codeconduct.rst]: done

**Comment by rcooke-ast** [comment]
I think this is a positive addition to the PypeIt developers community. One thing that is not immediately obvious to me is the definition of "community". It does sound more like this is for the developers (which is absolutely fine), but did you also want to mention something along the lines of:

"By using this software, we politely request that you abide by our code of conduct."

You might also consider adding something to this effect to the main `README.md` file (which is the first thing that users will likely read, and it will be displayed on the front page of the GitHub repo), and maybe put a link to the `codeofconduct.rst` file in the README.md file?

**Comment by profxj** [comment]
Merging

**Pull request merged**